source/common/upstream/health_checker_impl.cc

```diff
@@ -330,11 +339,17 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onTimeout() {
   }
 }
 
-Http::CodecClient::Type HttpHealthCheckerImpl::codecClientType(bool use_http2) {
-  if (use_http2) {
+Http::CodecClient::Type HttpHealthCheckerImpl::codecClientType(
+    const envoy::api::v2::core::HealthCheck::HttpHealthCheck::CodecClientType type) {
+  switch (type) {
+  case envoy::api::v2::core::HealthCheck::HttpHealthCheck::HTTP3:
+    return Http::CodecClient::Type::HTTP3;
+  case envoy::api::v2::core::HealthCheck::HttpHealthCheck::HTTP2:
     return Http::CodecClient::Type::HTTP2;
-  } else {
+  case envoy::api::v2::core::HealthCheck::HttpHealthCheck::HTTP1:
     return Http::CodecClient::Type::HTTP1;
+  default:
+    PANIC(fmt::format("Unknown codec client type {}", type));
```

Comment:
I think we should never get here due to validation so this can be NOT_REACHED?

Reply:
The input is from config, I think it's reachable if the config specifies an invalid enum value.

Reply:
You can make the config validate the the enum is only a defined value. Please do that.